### PR TITLE
[POAE7-1476]Update substraitRel and add DistributeRel to meet our requirements

### PR DESCRIPTION
**Pull request by ZJie1**

### CODE REVIEW

**Comment by ZJie1** [comment]
@Intel-bigdata/bdf-cider-all Please take a review.

File: binary/relations.proto [hunk 1]
```diff
@@ -88,6 +98,20 @@ message ReadRel {
 
 }
 
+message DistributeRel {
```
Comment (XinDongSh) [inline binary/relations.proto:101]: currently, the DistributeRel is not consistent with substrait doc(https://substrait.io/relations/physical_relations/#distribute-operator). Let's merge as it is for now and change it after rebase the substrait main branch.

**Review by XinDongSh — APPROVED** [review]

**Comment by winningsix** [comment]
Follow ups: 1. Merged; 2. Refine into three blockings 3. @ZJie1 to rebase.

**Comment by winningsix** [comment]
Please create a JIRA ID and with meaningful name. For this PR, it touched distributed rel. Please make it concrete.

File: binary/relations.proto [hunk 2]
```diff
@@ -17,14 +17,15 @@ message RelCommon {
...
     message Direct {}
     message Emit {
-        repeated int32 output_mapping = 1;
+        repeated Type.NamedStruct output_mapping = 1;
     }
 
     message Hint {
-        repeated HintKeyValue hint_key_values = 1;
+        repeated HintKeyValue hint_key_values= 1;
         Stats stats = 2;
```
Comment (yma11) [inline binary/relations.proto]: revert this, please.

**Pull request merged**